codegen/src/main/java/software/amazon/awssdk/codegen/emitters/tasks/EndpointProviderTasks.java

```diff
@@ -57,6 +59,9 @@ protected List<GeneratorTask> createTasks() throws Exception {
             tasks.add(generateDefaultProvider());
             tasks.add(new RulesEngineRuntimeGeneratorTask(generatorTaskParams));
         }
+        if (shouldGenerateJPathRuntime()) {
```

Comment:
What are the demerits of always loading the JMESPathRuntime class? Additionally, for waiters, what if we create a class that extends JMESPathRuntime and adds methods only specific to waiters?"

Reply:
The demerits of always loading JMESPathRuntime is that there are almost 400 services but only 80 of them use waiters. 

I don't think the approach of adding methods that are only used in waiters is worth it, because it's just a few hundred lines. It's better to keep the language intact and it will be hard to keep track of different subsets used for different purposes.

Reply:
Thanks